core/src/main/java/org/infinispan/Cache.java

```diff
@@ -443,4 +444,18 @@ default V compute(K key,
                      SerializableBiFunction<? super K, ? super V, ? extends V> remappingFunction) {
       return compute(key, (BiFunction<? super K, ? super V, ? extends V>) remappingFunction);
    }
+
+   /**
+    * {@inheritDoc}
+    * <p>
+    * When this method is used on a clustered cache, either replicated or distributed, the biconsumer will be serialized
+    * to owning nodes to perform the operation in the most performant way. However this means the biconsumer must
+    * have an appropriate {@link org.infinispan.commons.marshall.Externalizer} or be {@link java.io.Serializable} itself.
+    * <p>
+    * For transactional caches, whenever the values of the caches are collections, and the mapping function modifies the collection, the collection
```

Comment:
It's not specific to collections: the values in cache must not be mutated, period, and we'll rather enforce that than remove the limitation.